src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -160,6 +170,18 @@ protected virtual DynamoMigratorBase MigrateFrom(DynamoMigratorBase sourceMigrat
             Copy(sourceMigrator.DefinitionsDirectory, this.DefinitionsDirectory);
 
             this.PreferenceSettings = sourceMigrator.PreferenceSettings;
+            if (this.PreferenceSettings == null) return this;
+
+            // All preference settings are copied over including custom package folders
+            // However if one of the custom folder locations points to the user data directory
+            // of the previous version, it needs to be replaced with that of the current version
+            var indexToreplace = this.PreferenceSettings.CustomPackageFolders.FindIndex(f => f.Contains(sourceMigrator.UserDataDirectory));
+            
+            if (indexToreplace <= -1) return this;
```

Comment:
The notion of returning `this` is a little... strange. Perhaps this method can be made to return `void`? There's a slight chance, but I hope I didn't suggest returning `this` before :)